src/highrl/envs/teacher_env.py

```diff
@@ -55,6 +56,7 @@ def __init__(
         self.done: bool = False
 
         self.cfg = configure_teacher(teacher_config)
+        self.env_generator = PPO.load(self.cfg.generator_path)
```

Comment:
Avoid coupling please. Create a separate functions in `utils` for loading the model depending whether it's a reinforcement learning trained model or a supervised-trained model.